src/main/java/org/gradle/profiler/studio/StudioGradleClient.java

```diff
@@ -58,34 +68,40 @@ public StudioGradleClient(GradleBuildConfiguration buildConfiguration, Invocatio
         commandLine.add("-Xbootclasspath/a:" + Joiner.on(File.pathSeparator).join(launchConfiguration.getSharedJars()));
         commandLine.add(launchConfiguration.getMainClass());
         commandLine.add(invocationSettings.getProjectDir().getAbsolutePath());
+        System.out.println("* Android Studio logs can be found at: " + Paths.get(launchConfiguration.getStudioPluginsDir().toString(), "idea.log"));
         System.out.println("* Using command line: " + commandLine);
 
+        new StudioPluginInstaller().installPlugin(launchConfiguration);
         return new CommandExec().inDir(studioInstallDir.toFile()).start(commandLine);
     }
 
     @Override
     public void close() throws IOException {
-        System.out.println("* PLEASE STOP ANDROID STUDIO....");
+        System.out.println("* Stopping Android Studio....");
+        studioPluginConnection.send(new StudioRequest(EXIT));
         studioProcess.waitForSuccess();
         System.out.println("* Android Studio stopped.");
-
-        server.close();
+        studioPluginServer.close();
```

Comment:
Should the closing be done in a `finally` here or via some composite closeable to make sure all the servers have been stopped?

Reply:
Yeah, it makes sense